src/DIRAC/WorkloadManagementSystem/Client/Matcher.py

```diff
@@ -195,9 +195,19 @@ def _processResourceDescription(self, resourceDescription):
         resourceDict[name] = resourceDescription[name]
 
     if resourceDescription.get('Tag'):
```

Comment:
@andresailer What is the solution here? Should I revert these changes in `Matcher.py` and leave it to your PR?

Reply:
@atsareg wanted to investigate the source of the bad tags

Reply:
I can not reproduce "['MultiProcessor']" tag case but the empty tag should be fixed in #5385. Normally, tag should never be a string here

Reply:
So these fixes here are doing nothing but can be left just in case there is yet another case not spotted so far

Reply:
Contrary to my previous comment, empty string Tag can arrive from a pilot still, this will convert it to []. So, it is OK.

Reply:
I have to disgree with your assesment here

```python

    for name in multiValueMatchFields:  # includes 'Tag'
      if name in resourceDescription:
        resourceDict[name] = resourceDescription[name]   # Tag is now ''

    if resourceDescription.get('Tag'):  # isn't true for '', Tag is still empty string, not empty list.
```

https://github.com/DIRACGrid/DIRAC/blob/5d01dd1c5f04a01c9a68dae05f8abe63738c20f5/src/DIRAC/WorkloadManagementSystem/DB/TaskQueueDB.py#L32-L33

Reply:
Well, if resourceDescription.get('Tag') is empty string then there is no need to add it for the subsequent treatment to the resourceDict. Also RequiredTag (if any this would be an error) does not make sense in this case.